qdev_wrappers/fitting/Converter.py

```diff
@@ -0,0 +1,236 @@
+from qcodes.dataset.sqlite_base import connect
+from qdev_wrappers.file_setup import CURRENT_EXPERIMENT
+from qdev_wrappers.show_num import check_experiment_is_initialized
+
+
+def data_to_dict(id, samplefolder=None, datatype='SQL'):
+    if datatype == 'qcodes_legacy':
+        converter = Legacy_Converter()
+
+    if datatype == 'SQL':
+        converter = SQL_Converter()
+
+    return converter.convert(id, samplefolder)
+
+
+class DataConverter():
+    
+    def convert(self, id, samplefolder = None):
+        
+        data = self.find_data(id, samplefolder) 
+        data_dict = self.make_data_dictionary(data)
+
+        exp_id = self.find_experiment(id)
+        dependencies = self.find_dependencies(id, data)  #SQL needs only id number to find dependencies, Legacy needs only data
+        all_variables = self.find_variables(data)
+
+        data_dict['exp_id'] = exp_id
+        data_dict['run_id'] = id
+        data_dict['dependencies'] = dependencies
+        data_dict['variables'] = all_variables
+        
+        return data_dict
+    
+
+
+class SQL_Converter(DataConverter):
+    
+    def find_data(self, id, samplefolder):  
+        
+            '''sample folder does nothing except act as placeholder.                    
+            This function currently proceeds under the assumption that the experiment is set up, and doesn't
+            check anything to confirm that. Also, I think this doesn't return any error if it can't find 
+            the dataset with that id number.'''
+          
+            data = get_data_by_id(id)                   
+            
+            return data  
+
+        
+    def find_dependencies(self, id, data):      #SQL converter only needs id, not data
+        conn = connect('experiments.db')    #the name of the file to access can't be set manually right now 
+        cur = conn.cursor()
+        
+        names = {}
+        dependencies = {}
+
+        for row in cur.execute('SELECT layout_id, parameter FROM layouts WHERE run_id is ?', str(id)):
+            names[row[0]] = row[1]
+
+        for row in cur.execute('SELECT dependent, independent FROM dependencies'):
+            if row[0] in names.keys():
+                dependencies[names[row[0]]] = []
+            
+        for row in cur.execute('SELECT dependent, independent FROM dependencies'):     
+            if row[0] in names.keys():
+                dependencies[names[row[0]]].append(names[row[1]])
+
+        conn.close()
+
+        return dependencies
+
+    
+    def find_variables(self, data):
+        
+        all_variables = []
+        
+        for data_subset in data:
+            for variable in data_subset:
+                if variable['name'] not in all_variables:
+                    all_variables.append(variable['name'])
+                    
+        return all_variables
+
+    def find_experiment(self, id):
+
+        conn = connect('experiments.db')  # the name of the file to access can't be set manually right now
```

Comment:
again I think this is in the qc.config so should be possible to get it without it being hardcoded. It might also be in CURRENT_EXPERIMENT

Reply:
I would prefer going with the qcconfig rather than creating a dependency on CURRENT_EXPERIMENT